docs/cue.md

```diff
@@ -0,0 +1,59 @@
+CUE in Perses
+====================
+
+Perses comes with validation capabilities based on [CUE](https://cuelang.org/), a powerful validation language that permitted us to move the type constraints out of Golang (static language), which in the end makes it possible to modify at runtime the list of objects that Perses can accept. More concretely, this allows to support new kinds of charts and/or datasources dynamically, in the form of plugins written in CUE.
+
+# Writing plugins
+
+This section explains about the format any plugin should follow to be accepted & registered by Perses at runtime.
+
+## Chart
+
+A chart plugin looks like the following:
+
+```cue
+package line
+
+#panel: {
+	kind: "LineChart"
+	options: {
+		queries: [...#query]
+		show_legend?: bool
+	}
+}
+
+#query: _ // needed for make cue-test only
+```
+it should contain:
+- a package name (free text).
+- a `#panel` definition that holds:
+  - the chart's `kind`.
+  - an `options` map containing any attribute you want for this plugin.
+- `#query: _` to provide a mock value for `#query`
```

Comment:
@Nexucis I didnt take time to look into that but probably we want to improve the logic behind `make cue-test` here, rather than putting/asking plugins developer to put this dummy mock in the plugin definition.

Reply:
By the way does `make cue-test` failed with you provide a not known query ?

Reply:
it's mandatory as long as it's part of the definition, because since `make cue-test` evaluates each plugin independently, if no value is provided for `#query` within the plugin then the command returns an error "reference "#query" not found"